tensorflow_lite_support/c/test/task/vision/image_segmenter_test.cc

```diff
@@ -0,0 +1,334 @@
+constexpr float kGoldenMaskTolerance = 1e-2;
+// Magnification factor used when creating the golden category masks to make
+// them more human-friendly. Each pixel in the golden masks has its value
+// multiplied by this factor, i.e. a value of 10 means class index 1, a value of
+// 20 means class index 2, etc.
+constexpr int kGoldenMaskMagnificationFactor = 10;
+
+void InitializeColoredLabel(TfLiteColoredLabel& colored_label, uint8_t r,
+                            uint8_t g, uint8_t b, char* label) {
+  colored_label.r = r;
+  colored_label.g = g;
+  colored_label.b = b;
+  colored_label.label = strdup(label);
+  colored_label.display_name = nullptr;
+}
+
+TfLiteSegmentation CreatePartialDeepLabV3Segmentation() {
+  TfLiteSegmentation segmentation = {.width = 257, .height = 257};
+  segmentation.colored_labels = new TfLiteColoredLabel[21];
+  InitializeColoredLabel(segmentation.colored_labels[0], 0, 0, 0, "background");
+  InitializeColoredLabel(segmentation.colored_labels[1], 128, 0, 0,
+                         "aeroplane");
+  InitializeColoredLabel(segmentation.colored_labels[2], 0, 128, 0, "bicycle");
+  InitializeColoredLabel(segmentation.colored_labels[3], 128, 128, 0, "bird");
+  InitializeColoredLabel(segmentation.colored_labels[4], 0, 0, 128, "boat");
+  InitializeColoredLabel(segmentation.colored_labels[5], 128, 0, 128, "bottle");
+  InitializeColoredLabel(segmentation.colored_labels[6], 0, 128, 128, "bus");
+  InitializeColoredLabel(segmentation.colored_labels[7], 128, 128, 128, "car");
+  InitializeColoredLabel(segmentation.colored_labels[8], 64, 0, 0, "cat");
+  InitializeColoredLabel(segmentation.colored_labels[9], 192, 0, 0, "chair");
+  InitializeColoredLabel(segmentation.colored_labels[10], 64, 128, 0, "cow");
+  InitializeColoredLabel(segmentation.colored_labels[11], 192, 128, 0,
+                         "dining table");
+  InitializeColoredLabel(segmentation.colored_labels[12], 64, 0, 128, "dog");
+  InitializeColoredLabel(segmentation.colored_labels[13], 192, 0, 128, "horse");
+  InitializeColoredLabel(segmentation.colored_labels[14], 64, 128, 128,
+                         "motorbike");
+  InitializeColoredLabel(segmentation.colored_labels[15], 192, 128, 128,
+                         "person");
+  InitializeColoredLabel(segmentation.colored_labels[16], 0, 64, 0,
+                         "potted plant");
+  InitializeColoredLabel(segmentation.colored_labels[17], 128, 64, 0, "sheep");
+  InitializeColoredLabel(segmentation.colored_labels[18], 0, 192, 0, "sofa");
+  InitializeColoredLabel(segmentation.colored_labels[19], 128, 192, 0, "train");
+  InitializeColoredLabel(segmentation.colored_labels[20], 0, 64, 128, "tv");
+
+  return segmentation;
+}
+
+TfLiteSegmentation partial_deep_lab_v3_segmentation =
+    CreatePartialDeepLabV3Segmentation();
+
+// Checks that the two provided `TfLiteSegmentation`s are equal.
+void ExpectApproximatelyEqual(const TfLiteSegmentation& actual,
```

Comment:
Suggesting "ExpectPartiallyEqual" instead, it's not doing any approximations but just comparing a subset of the fields.

Reply:
Makes sense.